crates/miden-lib/src/account/auth/mod.rs

```diff
@@ -39,3 +42,69 @@ impl From<RpoFalcon512> for AccountComponent {
         .with_supports_all_types()
     }
 }
+
+/// An [`AccountComponent`] implementing a conditional RpoFalcon512 signature scheme for
+/// authentication of transactions.
+///
+/// This component only requires authentication when any of the specified procedures are called
+/// during the transaction. It stores a list of procedure roots that require authentication, and
+/// the signature verification is only performed if at least one of these procedures was invoked.
+///
+/// It exports the procedure `auth__tx_rpo_falcon512_conditional`, which:
+/// - Checks if any of the specified procedures were called during the transaction
+/// - If none were called, authentication is skipped
+/// - If at least one was called, performs standard RpoFalcon512 signature verification
+///
+/// The storage layout is:
+/// - Slot 0(value): Public key (same as RpoFalcon512)
+/// - Slot 1(value): Number of tracked procedures
+/// - Slot 2(map): A map with tracked procedure roots
+///
+/// This component supports all account types.
+pub struct RpoFalcon512ProcedureACL {
+    public_key: PublicKey,
+    trigger_procedures: Vec<Digest>,
+}
+
+impl RpoFalcon512ProcedureACL {
+    /// Creates a new [`RpoFalcon512ProcedureACL`] component with the given `public_key` and
+    /// list of procedure roots that require authentication.
+    ///
+    /// # Panics
+    /// Panics if more than 253 procedures are tracked (to leave room for the public key and count).
+    pub fn new(public_key: PublicKey, trigger_procedures: Vec<Digest>) -> Self {
+        assert!(
+            trigger_procedures.len() <= u8::MAX as usize - 2,
+            "Cannot track more than 253 procedures"
+        );
+        Self { public_key, trigger_procedures }
+    }
+}
+
+impl From<RpoFalcon512ProcedureACL> for AccountComponent {
+    fn from(conditional: RpoFalcon512ProcedureACL) -> Self {
+        let mut storage_slots = Vec::with_capacity(3);
+
+        // Slot 0: Public key
+        storage_slots.push(StorageSlot::Value(conditional.public_key.into()));
+
+        // Slot 1: Number of tracked procedures
+        let num_procs = Felt::from(conditional.trigger_procedures.len() as u32);
+        storage_slots.push(StorageSlot::Value([num_procs, Felt::ZERO, Felt::ZERO, Felt::ZERO]));
+
+        // Slots 2: A map with tracked procedure roots
+        for (i, proc_root) in conditional.trigger_procedures.iter().enumerate() {
+            storage_slots.push(StorageSlot::Map(
+                StorageMap::with_entries([(
+                    [Felt::ZERO, Felt::ZERO, Felt::ZERO, Felt::from(i as u32)].into(),
+                    proc_root.into(),
+                )])
+                .unwrap(),
+            ));
+        }
```

Comment:
This inserts a one-sized storage map per storage slot. The intention was to have a single storage map with multiple procedures, right?

Should we add a transaction-executing test for this component using >= 2 procedures?

Reply:
good catch, I clearly haven't given too much thought to the refactor from value -> map

Reply:
The test ended up unveiling some issues. Addressed in [fix: proper test and fixes](https://github.com/0xMiden/miden-base/pull/1531/commits/8e50684eea5363f654a40ba219851e9095a7b7e6)